pkg/apimachinery/patch/patch_test.go

```diff
@@ -0,0 +1,74 @@
+package patch
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("PatchSet", func() {
+
+	It("Patch remove operation", func() {
+		patch, err := New(
+			WithRemove("/abcd"),
+		).GeneratePayload()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(string(patch)).To(Equal(`[{"op":"remove","path":"/abcd"}]`))
```

Comment:
You can make this one-liner:
```go
Expect(New(WithRemove("/abcd")).GeneratePayload()).To(
  Equal([]byte(`[{"op":"remove","path":"/abcd"}]`)),
)
```

gimkgo/gomega checks implicitly if the 2nd return value is non-nil and will fail on it.

For example:
```go
Expect(func() (string, error) { return "hello", fmt.Errorf("myerr") }()).To(Equal("hello"))
```
fails with:
```
  [FAILED] Unexpected error: myerr
      <*errors.errorString | 0xc00033a470>: 
      myerr
      {s: "myerr"}
```

I like to use it when it is possible, it makes things shorter.

Reply:
@EdDev the only that makes is a bit hard to put everything into a single line is that `GeneratePayload()` return []byte and we need to compare against a string. Converting the patch string to []bytes seems a bit weird to me

Reply:
The error by converting the expected value to []bytes looks like:
```
  [FAILED] Expected
      <[]uint8 | len:48, cap:48>: [{"op":"replace","path":"/abcd","value":"test"}]
  to equal
      <[]uint8 | len:49, cap:64>: [{"op":"replace","path":"/abcd","values":"test"}]
```

Reply:
`value` doesn't match `values`?

Reply:
@0xFelix sorry I don't understand the question. In any case, I'm using []byte not string anymore